src/Tes.ApiClients/TerraSamApiClient.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using CommonUtilities;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Tes.ApiClients.Models.Terra;
+
+namespace Tes.ApiClients
+{
+    /// <summary>
+    /// Terra Sam api client
+    /// Sam manages authorization and IAM functionality
+    /// </summary>
+    public class TerraSamApiClient : TerraApiClient
+    {
+        private const string SamApiSegments = @"/api/azure/v1";
+
+        // TODO configure cache and turn caching of queries back on
+        private static readonly IMemoryCache SharedMemoryCache = new MemoryCache(new MemoryCacheOptions());
```

Comment:
Ideally we would cache these responses but for no more than a guaranteed short time window (5 minutes?). Does that feel like a reasonable expectation for this caching library? Or should we make a request every time if we want to guarantee a certain level of freshness?

Reply:
A 5-minute cache time seems perfectly reasonable to me.